src/widget/wknobcomposed.h

```diff
@@ -28,6 +28,10 @@ class WKnobComposed : public WWidget {
     void mouseReleaseEvent(QMouseEvent *e) override;
     void paintEvent(QPaintEvent* /*unused*/) override;
 
+    bool m_defaultAngleIsValid;
+    double m_defaultAngle;
```

Comment:
if you wrap this in an optional, its impossible to accidentally use it without checking.

Reply:
so `std::optional<double> m_defaultAngle;` would not require ..isValid anymore, we'd just check
`if (m_defaultAngle) { .. } else ..`

Reply:
No, I'm not getting it.
(your comment in _this_ place)

I'll try using std::optional<double> for neutralPointOnScale()

Reply:
yes, though its often nicer to use `value_or` and the monadic operations added in C++23. Id also prefer `has_value` over `operator bool` in if conditions. [`std::optional<T>::operator->, std::optional<T>::operator*`](https://en.cppreference.com/w/cpp/utility/optional/operator*) should be avoided!

Reply:
ah okay, I was confused why I could use m_angle right away, std::optional<T> has `value()`

Reply:
yeah. the difference is that one is checked the other is not. The problem with that API is that the check could now be very far away from the use and could get lost in a future refactoring. `value()` is safer (in the sense that it `std::terminates` when there is no value, instead of UB) but obviously checks each and every time. optional monadic operations `C++23` solve that.